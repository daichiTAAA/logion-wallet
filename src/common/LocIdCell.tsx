import React from 'react';

import { UUID } from '../logion-chain/UUID';

import { Cell } from './Table';
import { LocRequestStatus } from './types/ModelTypes';

export interface Props {
    status: LocRequestStatus;
    id: string;
}

export default function LocIdCell(props: Props) {

    let content: string;
    if(props.status === 'OPEN' || props.status === 'CLOSED') {
        content = new UUID(props.id).toDecimalString();
    } else {
        content = "-";
    }

    return (
        <Cell
            overflowing
            tooltipId={ `locId-${ props.id }` }
            content={ content }
        />
    );
}
