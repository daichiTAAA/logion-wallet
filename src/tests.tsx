import { ReactElement } from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import reactRenderer, { ReactTestRenderer, ReactTestRendererJSON, act as reactAct } from 'react-test-renderer';

export function render(element: ReactElement): null | ReactTestRendererJSON | ReactTestRendererJSON[] {
    const component = reactRenderer.create(element);
    return component.toJSON();
}

export function renderer(element: ReactElement): ReactTestRenderer {
    return reactRenderer.create(element);
}

export function shallowRender<E extends ReactElement>(element: ReactElement): E {
    const renderer = createRenderer();
    renderer.render(element);
    return renderer.getRenderOutput();
}

export function mockAccount(address: string, name: string) {
    return {
        address,
        meta: {
            name
        }
    };
}

export { reactAct as act };
