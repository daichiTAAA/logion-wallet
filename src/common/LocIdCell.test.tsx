import { shallowRender } from '../tests';
import LocIdCell from './LocIdCell';

it("renders UUID as decimal", () => {
    const tree = shallowRender(<LocIdCell
        id='6ec0bd7f-11c0-43da-975e-2a8ad9ebae0b'
    />);
    expect(tree).toMatchSnapshot();
});
