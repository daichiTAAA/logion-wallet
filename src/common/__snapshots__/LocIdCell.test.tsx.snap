// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders UUID as decimal 1`] = `
<Cell
  content="147215843976064841756558269764787219979"
/>
`;
